include/aspect/stokes_matrix_free.h

```diff
@@ -306,8 +306,21 @@ namespace aspect
       virtual void setup_dofs()=0;
 
       /**
```

Comment:
I had to move `evaluate_material_model()` and `correct_stokes_rhs()` to be public functions since I now need to call them separate from the `build_preconditioner()` in `solver_schemes.cc` and `assembly.cc`. Also, now `build_preconditioner()` is the same as the old `compute_A_block_diagonals()`.

Reply:
I like the new naming.